src/modules/Pam.rb

```diff
@@ -25,14 +25,43 @@
 # Authors:	Jiri Suchomel <jsuchome@suse.cz>
 # Flags:	Unstable
 #
-# $Id$
-#
 require "yast"
+require "yast2/execute"
 
 module Yast
   class PamClass < Module
+    include Yast::Logger
+
     def main
+    end
+
+    PAM_CONFIG_BIN = "/usr/sbin/pam-config".freeze
+
+    # List PAM modules.
+    #
+    # The returned module names do not include the "pam_" prefix nor
+    # the ".so" suffix. So just "pwquality" instead of "pam_pwquality.so".
+    #
+    # @return [Array{String}]
+    def List
+      begin
+        lines = Yast::Execute.locally!(PAM_CONFIG_BIN, "--list-modules",
+          :stdout => :capture).split("\n")
+      rescue Cheetah::ExecutionFailed => e
+        log.error "pam-config --list-modules failed #{e.message}"
+        return []
+      end
+
+      rx = /[ \t]*pam_([a-z0-9_]+)\.so[ \t]*/
```

Comment:
np: I would move the regular expression to a constant.

Reply:
Just make it uppercase? Or local to the module?

Reply:
Local to the module. You can put above the method where is used:

```ruby
RX = /.../
private_constant :RX

def List
# ...
end
```